fix(api): Limit how far up the gripper moves on the Z when opening/closing the plate reader lid.

vegano1 opened the pull request:
# Overview

The Flex has a new border around the stacker window that gets hit by the plate reader lid when opening/closing the lid. This pull request extends the `get_gripper_labware_movement_waypoints` so you can optionally add a z offset to limit how far up the Z the gripper travels when moving labware. We then use this to limit the z movement of the plate reader lid in the `open_lid` and `close_lid` implementations of the plate reader commands.

Closes: [RQA-4439](https://opentrons.atlassian.net/browse/RQA-4439)


https://github.com/user-attachments/assets/4b8cca13-2bcd-4243-a8dc-3debcc1de4ed



## Test Plan and Hands on Testing

- [x] Make sure the plate reader no longer hits the side if the flex lip when `open_lid` and `close_lid` are used

## Changelog

- Extend `get_gripper_labware_movement_waypoints` so you can optionally add a z offset to limit how far up the Z the gripper travels when moving labware.
- Limit the z movement of the plate reader lid to 40mm + 16mm (max lw height in stacker) in the `open_lid` and `close_lid` commands.

## Review requests
## Risk assessment

[RQA-4439]: https://opentrons.atlassian.net/browse/RQA-4439?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

CaseyBatten committed:
initial plate reader lid lift heigh adjustment

vegano1 committed:
Merge branch 'edge' into rqa-4439_plate_reader_lid_lift_height_adjustment

vegano1 committed:
feat(api): limit how far up the gripper moves up when opening/closing the plate reader lid so it does not hit the new PVT flex window lip.

sfoster1 reviewed (approved):
Looks good to me, thank you!

ahiuchingau reviewed (approved):
Nice, LGTM

github-actions[bot] committed:
fix(analyses-snapshot-testing): heal rqa-4439_plate_reader_lid_lift_height_adjustment snapshots (#19120)

This PR was requested on the PR
https://github.com/Opentrons/opentrons/pull/19118

Co-authored-by: y3rsh <502770+y3rsh@users.noreply.github.com>
